### PR TITLE
chore(ssa refactor): Add code to handle assert statement in acir gen

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

I'd like to hold off on merging this in because I want to:

- Tidy up the massive doc comment on `is_equal` 
- Open up a separate PR for the new `to_expression` method added to AcirVarData which may be able to clean up a few methods. 
- Open up an issue regarding the fact that using eq_var for a constrain statement is in-efficient. This is the same with our old codebase and there are quite a few ways to handle this, one of them being for Acir gen to do a pre-scan to check which values are being used for constrain statements, another being to do the simplification pass and another being for the SSA IR or AST to explicitly handle a constrain statement as one instruction. (Even if we stick with the current strategy, it would be good to write down the rationale for this in the issue for later documentation) 

This code will handle the assert statement for acir gen

### CODE REVIEW

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs [hunk 1]
```diff
@@ -300,6 +312,15 @@ impl AcirVarData {
         }
         None
     }
+    /// Converts all enum variants to an Expression.
+    /// TODO: can refactor code in this file to use this
```
Comment (joss-aztec) [inline crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs:316]: Want an issue for this?

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs [hunk 2]
```diff
@@ -118,6 +124,97 @@ impl GeneratedAcir {
...
+    ///
+    /// Intuition: the equality of two Expressions is linked to whether
+    /// their difference has an inverse. a == b implies that a - b == 0
+    /// which implies that a - b has no inverse. So if two variables are equal
+    /// their difference will have no inverse.
+    ///
+    /// First, lets create a new variable that is equal to the difference
+    /// of the two expressions: `t = lhs - rhs` (constraint has been applied)
+    ///
+    /// Next lets create a new variable `y` which will the Witness, we will ultimately
```
Comment (joss-aztec) [inline crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/generated_acir.rs:139]:
```suggestion
    /// Next lets create a new variable `y` which will be the Witness that we will ultimately
```

File: crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs (hunk 1 above)
Comment (kevaundray) [reply crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs:316]: Created #1441

**Comment by kevaundray** [comment]
Opened up #1442 regarding how we process equals within a constrain

**Review by joss-aztec — APPROVED** [review]

**Pull request merged**